eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxos.hs

```diff
@@ -182,6 +188,23 @@ scriptsValidateTransition = do
         totalDeposits pp (`Map.notMember` poolParams) txcerts <-> refunded
   sysSt <- liftSTS $ asks systemStart
   ei <- liftSTS $ asks epochInfo
+  let ~traceIdentifier =
+        (castHash $ hashWithSerialiser @ShortHash toCBOR slot)
+          `xor` (castHash $ hashWithSerialiser @ShortHash toCBOR txb)
+          `xor` (castHash $ hashWithSerialiser @ShortHash toCBOR poolParams)
+          `xor` (castHash $ hashWithSerialiser @ShortHash toCBOR genDelegs)
+          `xor` (castHash $ hashWithSerialiser @ShortHash toCBOR deposited)
+          `xor` (castHash $ hashWithSerialiser @ShortHash toCBOR utxo)
```

Comment:
This seems like an awful lot of hashing - for what reason? Is this identifier going to be something consumers of this event will actually use?

Reply:
And on second look, this hashes the UTxO!

Reply:
yes, this is just for the GHC event log. it's important to be able to correlate the "BEGIN"/"END" events so that we have useful information about how long script validation/execution takes. this shouldn't be evaluated without the appropriate rts flag on. I just removed hashing the utxo since maybe the txbody is enough

Reply:
From the `Debug.Trace` module:
> Note that the string message is always evaluated, whether or not profiling is available or enabled.

So this will be evaluated for every transaction. The ledger is single-threaded, so correlation shouldn't be an issue.

I would suggest, if anything, just using the txbody hash. This is at least already serialised. However, I'd question whether we really need this, since events should not be interleaved.